ros2cli/ros2cli/helpers.py

```diff
@@ -118,3 +120,18 @@ def collect_stdin():
             break
         lines += line
     return lines
+
+
+def get_rmw_additional_env(rmw_implementation: str) -> Dict[str, str]:
```

Comment:
I really think that `rmw_implementation` should handle those configuration but `ros2cli`.
again, i do not want to block this zenoh development, but i am not convinced to take this approach with following software design...

- Separation of Concerns: By moving the logic to the RMW API, we can maintain a clear separation between the middleware-specific logic and the `ros2cli`. This makes the codebase cleaner and easier to maintain.
- Single Source of Truth: Having the environment variable logic centralized in the RMW API ensures that any changes or updates to the middleware's environment variables are made in one place. This reduces the risk of inconsistencies.
- Reusability: Other tools or components that need the same environment variable information can reuse the RMW API.

Reply:
@fujitatomoya these are excellent points. Our goal is not not bias the various ROS 2 repos towards `rmw_zenoh` by having this be a permanent change. We will definitely revert this change before Kilted is released but changes like these are needed to help test rmw_zenoh with the existing infra while we upgrade infra in parallel to support starting the router. In fact, @cottsay is working on additions to the RMW API to avoid the changes needed here.
I'll add an agenda item for tomorrow's PMC meeting to discuss this in greater length.